include/boost/geometry/strategies/cartesian/side_of_intersection.hpp

```diff
@@ -27,6 +35,104 @@ namespace boost { namespace geometry
 namespace strategy { namespace side
 {
 
+namespace detail
+{
+
+template <typename T, T N, std::size_t P>
+struct integral_pow
```

Comment:
Yes, I think you're right that it should be possible to implement a version with lesser complexity. I also agree that this is not important as long as we don't plan to use it anywhere else in the library. If needed it could be moved e.g. to `util` and refactored.